diff --git a/node_modules/react-native/React/CoreModules/React-CoreModules.podspec b/node_modules/react-native/React/CoreModules/React-CoreModules.podspec
index d41b66c..76d1aff 100644
--- a/node_modules/react-native/React/CoreModules/React-CoreModules.podspec
+++ b/node_modules/react-native/React/CoreModules/React-CoreModules.podspec
@@ -49,7 +49,7 @@ Pod::Spec.new do |s|
   s.ios.exclude_files      = "PlatformStubs/**/*"
   exclude_files            = ["RCTStatusBarManager.mm"]
   s.macos.exclude_files    = exclude_files
-  s.visionos.exclude_files = exclude_files
+  s.visionos.exclude_files = exclude_files if s.respond_to?(:visionos)
   s.tvos.exclude_files     = exclude_files
 
   s.header_dir             = "CoreModules"
diff --git a/node_modules/react-native/sdks/hermes-engine/hermes-engine.podspec b/node_modules/react-native/sdks/hermes-engine/hermes-engine.podspec
index 326c6fa..e59ed3f 100644
--- a/node_modules/react-native/sdks/hermes-engine/hermes-engine.podspec
+++ b/node_modules/react-native/sdks/hermes-engine/hermes-engine.podspec
@@ -48,7 +48,7 @@ Pod::Spec.new do |spec|
   spec.ios.vendored_frameworks = "destroot/Library/Frameworks/ios/hermes.framework"
   spec.tvos.vendored_frameworks = "destroot/Library/Frameworks/tvos/hermes.framework"
   spec.osx.vendored_frameworks = "destroot/Library/Frameworks/macosx/hermes.framework"
-  spec.visionos.vendored_frameworks = "destroot/Library/Frameworks/xros/hermes.framework"
+  spec.visionos.vendored_frameworks = "destroot/Library/Frameworks/xros/hermes.framework" if spec.respond_to?(:visionos)
 
   if HermesEngineSourceType::isPrebuilt(source_type) then
 
@@ -57,7 +57,7 @@ Pod::Spec.new do |spec|
       ss.source_files = "destroot/include/hermes/**/*.h"
       ss.header_mappings_dir = "destroot/include"
       ss.ios.vendored_frameworks = "destroot/Library/Frameworks/universal/hermes.xcframework"
-      ss.visionos.vendored_frameworks = "destroot/Library/Frameworks/universal/hermes.xcframework"
+      ss.visionos.vendored_frameworks = "destroot/Library/Frameworks/universal/hermes.xcframework" if ss.respond_to?(:visionos)
       ss.tvos.vendored_frameworks = "destroot/Library/Frameworks/universal/hermes.xcframework"
       ss.osx.vendored_frameworks = "destroot/Library/Frameworks/macosx/hermes.framework"
     end
